src/System.AppContext/src/System/AppContext.cs

```diff
@@ -18,8 +18,7 @@ private enum SwitchValueState
             UnknownValue = 0x8 // Not used on .NET Native
         }
 
-        private static Dictionary<string, SwitchValueState> s_switchMap = new Dictionary<string, SwitchValueState>();
-        private static readonly object s_syncLock = new object();
+        private static readonly Dictionary<string, SwitchValueState> s_switchMap = new Dictionary<string, SwitchValueState>();
```

Comment:
No, because it still has "found true", "found false" and "not found" as the possible outcomes, which just stashing the key alone doesn't provider for.